.github/workflows/release-sync.yml

```diff
@@ -0,0 +1,162 @@
+
+      - name: Create Pull Request
+        id: create_pull
+        run: |
+          # The assignee will be the user that manually published the release
+          PR_URL=$(gh pr create \
+            --title "chore: [${{ github.ref_name }}] Merge release into master" \
+            --body "Automated PR to merge `release` branch into `master` based on release event." \
+            --base "master" \
+            --head "release" \
+            --assignee "@me")
+
+          # Store value in step output
+          echo "PR_URL=$PR_URL" >> "$GITHUB_OUTPUT"
+
+      - name: Associate PR with project
+        env:
+          PR_URL: ${{ steps.create_pull.outputs.PR_URL }}
+        run: |
+          gh project item-add $PROJECT_NUMBER \
+            --owner $ORG_NAME \
+            --url "$PR_URL" \
+
+      - name: Fetch project and column data
+        id: proj_columns
+        env:
+          PR_URL: ${{ steps.create_pull.outputs.PR_URL }}
+        # There is no direct command to move a card inside a project. Here we start gathering id data about the project
+        # to do this operation through the API
+        run: |
+          QUERY="
+          query {
+            organization(login: \"$ORG_NAME\") {
+              projectV2(number: $PROJECT_NUMBER) {
+                id
+                field(name: \"Status\") {
+                  __typename
+                  ...on ProjectV2SingleSelectField {
+                    id
+                    options {
+                      id
+                      name
+                    }
+                  }
+                }
+              }
+            }
+          }
+          "
+          PROJ_RESPONSE=$(gh api graphql -f query="$QUERY")
+
+          # Parse json response and fetch necessary data identifiers from it
+          PROJ_DATA=$(echo "$PROJ_RESPONSE" | jq -r --arg column_name "$COLUMN_NAME" '
+            .data.organization.projectV2.id as $projectId |
+            .data.organization.projectV2.field.id as $fieldId |
+            .data.organization.projectV2.field.options[] |
+            select(.name == $column_name) |
+            {
+              projectId: $projectId,
+              fieldId: $fieldId,
+              optionId: .id
+            }')
+          PROJECT_ID=$(echo "$PROJ_DATA" | jq -r '.projectId')
+          FIELD_ID=$(echo "$PROJ_DATA" | jq -r '.fieldId')
+          OPTION_ID=$(echo "$PROJ_DATA" | jq -r '.optionId')
+
+          # Store values in step output
+          echo "PROJECT_ID=$PROJECT_ID" >> "$GITHUB_OUTPUT"
+          echo "FIELD_ID=$FIELD_ID" >> "$GITHUB_OUTPUT"
+          echo "OPTION_ID=$OPTION_ID" >> "$GITHUB_OUTPUT"
+
+      - name: Fetch PR card Id
+        id: proj_cardId
+        env:
+          PR_URL: ${{ steps.create_pull.outputs.PR_URL }}
+        run: |
+          QUERY="
+            query {
+              organization(login: \"$ORG_NAME\") {
+                projectV2(number: $PROJECT_NUMBER) {
+                  items(first: 10) {
```

Comment:
Will this still work in case there are more than 10 cards in the column, and the one you're looking for is not among the first 10?

Reply:
That's a very important point. Our main project currently has just under 400 items, and that query surely wouldn't work.

I'll look into a way of restricting the query only for the desired repository.

Reply:
I fixed this by using another approach to the query: instead of querying the project for the specific PR, I queried the PR for its related project item. This was done in https://github.com/HathorNetwork/hathor-wallet-lib/pull/692/commits/e7a592034a628f64e7cdc1a127dfff78cf61130a